src/hotspot/share/nmt/vmatree.cpp

```diff
@@ -62,6 +62,81 @@ VMATree::SummaryDiff VMATree::register_mapping(position A, position B, StateType
+    //   4  Reserved   Committed     ..       committing a reserved region             OK                            public     commit_mapping
+    //   5  Committed     ..         ..       re-committing a region                   OK                            public         ..
+    //   6  Released      ..         ..       committing a non-reserved region         invalid mapping request       public         ..
+    //   7  Reserved   Released      ..       releasing a reserved region              invalid mapping request       private    register_mapping
+    //   8  Committed     ..         ..       releasing a committed region             invalid mapping request       private        ..
+    //   9  Released      ..         ..       re-releasing a region                    ?                             private        ..
+    //  10  Reserved   Reserved     false     re-reserving a region                    split a reserved region       public     reserve_mapping
+    //  11  Committed     ..         ..       uncommitting a region                    Error                         public         ..
+    //  12  Released      ..         ..       reserving a released region              OK                            public         ..
+    //  13  Reserved   Committed     ..       committing a reserved ergion             OK                            public     commit_mapping
+    //  14  Committed     ..         ..       re-committing a region                   flags should match            public         ..
+    //  15  Released      ..         ..       committing a non-reserved region         invalid mapping request       public         ..
+    //  16  Reserved   Released      ..       releasing a reserved region              OK                            private    register_mapping
+    //  17  Committed     ..         ..       releasing w/out uncommitting a region    OK                            private        ..
+    //  18  Released      ..         ..       re-releasing a region                    OK                            private        ..
+
+
+    static int count_6 = 0;
+    static int count_11 = 0;
+    static int count_14 = 0;
+    auto notify = [&](int row_no, const char* str) -> bool {
+      if (row_no == 6 || row_no == 11 || row_no == 14) {
+        if (row_no == 6)  { if (count_6++) return true; }
+        if (row_no == 11) { if (count_11++) return true; }
+        if (row_no == 14) { if (count_14++) return true; }
+        tty->print_cr("new region " SIZE_FORMAT "-" SIZE_FORMAT, (size_t)A, (size_t)B);
+        tty->print_cr("leqA node " SIZE_FORMAT ", flag: %d", (size_t)leqA_n->key(), (int)leqA_n->val().out.flag());
+        warning("%s", str);
+        return true;
+      } else {
+        fatal("%s", str);
+        return false;
+      }
+      return false;
+    };
+    VMATree::SummaryDiff invalid_diff(-1);
```

Comment:
Can we replace hardcoded `-1` with some a constant in:

`VMATree::SummaryDiff invalid_diff(-1);`

and what is the caller supposed to do with that? How would it check for the special INVALID_DIFF value?

For example, in `src/hotspot/share/nmt/nmtNativeCallStackStorage.hpp` we define:

`constexpr static const StackIndex invalid = std::numeric_limits<StackIndex>::max() - 1;`

and offer the API:

`static bool is_invalid(StackIndex a)`